compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -480,14 +480,17 @@ impl<'a> Linker for GccLinker<'a> {
     fn control_flow_guard(&mut self) {}
 
     fn debuginfo(&mut self, strip: Strip) {
+        // MacOS linker doesn't support stripping symbols directly anymore.
+        if self.sess.target.is_like_osx {
+            return;
+        }
+
         match strip {
             Strip::None => {}
             Strip::Debuginfo => {
-                // MacOS linker does not support longhand argument --strip-debug
                 self.linker_arg("-S");
             }
             Strip::Symbols => {
-                // MacOS linker does not support longhand argument --strip-all
                 self.linker_arg("-s");
             }
```

Comment:
If MacOS is not taking this code path anymore, it's probably worth reverting to longhand arguments for clarity.

Reply:
That makes a lot of sense. I've pushed an update with them.